src/toniarts/openkeeper/tools/convert/KmfModelLoader.java

```diff
@@ -490,28 +502,20 @@ private VertexBuffer[] createIndices(final Map<Integer, List<Triangle>> triangle
         // Ultimately all failed, now just instead off completely empty buffers, put one 0 there, seems to have done the trick
         //
         // Triangles, we have LODs here
-        VertexBuffer[] lodLevels = new VertexBuffer[trianglesMap.size()];
-        for (Entry<Integer, List<Triangle>> triangles : trianglesMap.entrySet()) {
-            if (!triangles.getValue().isEmpty()) {
-                int[] indexes = new int[triangles.getValue().size() * 3];
+        var lodLevels = new VertexBuffer[trianglesMap.size()];
+        for (var triangles : trianglesMap.entrySet()) {
+            var buf = new VertexBuffer(Type.Index);
+            var indexes = new short[triangles.getValue().size() * 3];
+
                 int x = 0;
                 for (Triangle triangle : triangles.getValue()) {
                     indexes[x * 3] = triangle.getTriangle()[2];
                     indexes[x * 3 + 1] = triangle.getTriangle()[1];
                     indexes[x * 3 + 2] = triangle.getTriangle()[0];
-                    x++;
+                ++x;
                 }
-                VertexBuffer buf = new VertexBuffer(Type.Index);
-                buf.setupData(VertexBuffer.Usage.Static, 3, VertexBuffer.Format.UnsignedInt, BufferUtils.createIntBuffer(indexes));
+            buf.setupData(VertexBuffer.Usage.Static, 3, VertexBuffer.Format.UnsignedShort, BufferUtils.createShortBuffer(indexes));
```

Comment:
The original data is even in bytes, not sure why it's saved as shorts.
https://github.com/tonihele/OpenKeeper/blob/995a56ea9e3b2eabe02614ff676fec4a7e33fe3f/src/toniarts/openkeeper/tools/convert/kmf/KmfFile.java#L340-L342

Reply:
Oh yeah, sorry. I think I answered in the other PR just now. But yeah, lets put it here too. We read everything as one size bigger than the original to count for the unsigned stuff. That is how we get correct value to the file format objects. How it is then used in our conversions... that is another thing.